modules/redis/src/main/scala/zio/redis/SingleNodeSubscriptionExecutor.scala

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2021 John A. De Goes and the ZIO contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.redis
+
+import zio.redis.Input.{CommandNameInput, StringInput}
+import zio.redis.Output.PushProtocolOutput
+import zio.redis.SingleNodeSubscriptionExecutor.{Request, RequestQueueSize, True}
+import zio.redis.api.Subscription
+import zio.redis.internal._
+import zio.redis.options.PubSub.PushProtocol
+import zio.stream._
+import zio.{Chunk, ChunkBuilder, IO, Promise, Queue, Ref, Schedule, Scope, URIO, ZIO}
+
+private[redis] final class SingleNodeSubscriptionExecutor private (
+  channelSubsRef: Ref[Map[String, Chunk[Queue[Take[RedisError, PushProtocol]]]]],
+  patternSubsRef: Ref[Map[String, Chunk[Queue[Take[RedisError, PushProtocol]]]]],
```

Comment:
Can't we use hubs?

Reply:
This is to ensure the callback is invoked. response messages can be delivered from Redis before target streams are consumed. if we use hub then response messages can be loss that have to invoke callbacks

Reply:
but maybe It seems like we could use a promise queue for handling response messages (like `SingleNodeExecutor`) to enable using the hub 🤔

Reply:
I added promise queue for waiting subscribe responses and let them use `Hub` intead of chunks of `Queue`